File: nim/src/call_coding_clis/prompt_spec.nim
import std/strutils
import std/options
import std/strtabs
import call_coding_clis/runner

proc buildPromptSpec*(prompt: string): CommandSpec =
  let trimmed = prompt.strip()
  if trimmed.len == 0:
    raise newException(ValueError, "prompt must not be empty")
  result = CommandSpec(
    argv: @["opencode", "run", trimmed],
    stdinText: none(string),
    cwd: none(string),
    env: newStringTable()
  )

File: elixir/lib/call_coding_clis/prompt_spec.ex
defmodule CallCodingClis.PromptSpec do
  alias CallCodingClis.CommandSpec

  def build(prompt) when is_binary(prompt) do
    trimmed = String.trim(prompt)

    if trimmed == "" do
      raise ArgumentError, "prompt must not be empty"
    end

    %CommandSpec{argv: ["opencode", "run", trimmed]}
  end
end
